examples/interfaces.rs

```diff
@@ -0,0 +1,191 @@
+
+            rt.block_on(del_interface(handle, if_name))
+        }
+        None | Some(_) => bail!("Must specify op: <new> or <del>"),
+    }
+}
+
+async fn new_interface(
+    handle: Nl80211Handle,
+    wiphy_name: String,
+    if_name: String,
+    if_type: Nl80211InterfaceType,
+) -> anyhow::Result<()> {
+    let mut phy_handle = handle.wireless_physic().get().execute().await;
+
+    let mut wiphy_id = None;
+    // There's surely a better way to do this, but TryStream's API put up a good
+    // fight
+    while let Some(msg) = phy_handle.try_next().await? {
+        if let Nl80211Message {
+            cmd: Nl80211Command::NewWiphy,
+            attributes,
+        } = msg.payload
+        {
+            let opt = find_attr(
+                &attributes,
+                |a| {
+                    if let Nl80211Attr::WiphyName(n) = a {
+                        n == &wiphy_name
+                    } else {
+                        false
+                    }
+                },
+                |a| {
+                    if let Nl80211Attr::Wiphy(id) = a {
+                        Some(*id)
+                    } else {
+                        None
+                    }
+                },
+            );
+            wiphy_id = wiphy_id.or(opt);
+        }
+    }
+    let wiphy_id = wiphy_id
+        .ok_or_else(|| anyhow!("Could not find wiphy <{}>", wiphy_name))?;
+
+    debug!("Using wiphy id {wiphy_id}");
+    let mut if_handle = handle
+        .interface()
+        .add(vec![
+            Nl80211Attr::Wiphy(wiphy_id),
+            Nl80211Attr::IfType(if_type),
+            Nl80211Attr::IfName(if_name),
+        ])
+        .execute()
+        .await;
+
+    while let Some(msg) = if_handle.try_next().await? {
+        info!("Response: {msg:?}")
+    }
+
+    Ok(())
+}
+
+async fn del_interface(
+    handle: Nl80211Handle,
+    if_name: String,
+) -> anyhow::Result<()> {
+    // no attributes = all interfaces
+    let mut if_handle = handle.interface().get(vec![]).execute().await;
+
+    let mut dump_handle = None;
+    while let Some(msg) = if_handle.try_next().await? {
+        if let Nl80211Message {
+            cmd: Nl80211Command::NewInterface,
+            attributes,
+        } = msg.payload
+        {
+            let opt = find_attr(
+                &attributes,
+                |a| {
+                    if let Nl80211Attr::IfName(n) = a {
+                        n == &if_name
+                    } else {
+                        false
+                    }
+                },
+                |a| {
+                    if let Nl80211Attr::IfIndex(id) = a {
+                        Some(*id)
+                    } else {
+                        None
+                    }
+                },
+            );
+            dump_handle = dump_handle.or(opt);
+        }
+    }
+    let if_index = dump_handle
+        .ok_or_else(|| anyhow!("Could not find interface <{}>", if_name))?;
```

Comment:
Some wild alternative implementation could be this

```suggestion
    let mut if_attributes = std::pin::pin!(handle
        .interface()
        .get(vec![])
        .execute()
        .await
        .try_filter_map(|msg| async move {
            match msg.payload {
                Nl80211Message {
                    cmd: Nl80211Command::NewInterface,
                    attributes,
                } if                     attributes
                    .iter()
                    .any(|f| matches!(f, wl_nl80211::Nl80211Attr::IfName(name) if name == if_name))
                 => Ok(Some(attributes)),
                _ => Ok(None),
            }
        }));

    let if_index = *if_attributes
        .try_next()
        .await?
        .ok_or_else(|| anyhow!("Could not find interface <{}>", if_name))?
        .iter()
        .find_map(|attr| match attr {
            Nl80211Attr::IfIndex(i) => Some(i),
            _ => None,
        })
        .unwrap();

```